Core/automation/lib/python/core/metadata.py

```diff
@@ -0,0 +1,235 @@
+"""
+This module provides functions for manipulating Item Metadata.
+
+
+"""
+
+__all__ = [
+    "get_all_namespaces", "get_metadata", "set_metadata", "remove_metadata",
+    "get_value", "set_value", "get_key_value", "set_key_value",
+    "remove_key_value"
+]
+
+from collections import MutableMapping
+import re
+
+from core import osgi
+from core.jsr223.scope import itemRegistry
+from core.utils import validate_item
+
+try:
+    from org.openhab.core.items import Metadata, MetadataKey
+    from org.openhab.core.items import ItemNotFoundException
+except:
+    from org.eclipse.smarthome.core.items import Metadata, MetadataKey
+    from org.eclipse.smarthome.core.items import ItemNotFoundException
+
+metadata_registry = osgi.get_service(
+        "org.openhab.core.items.MetadataRegistry"
+    ) or osgi.get_service(
+        "org.eclipse.smarthome.core.items.MetadataRegistry"
+    )
+
+from core.log import logging, LOG_PREFIX
+log = logging.getLogger("{}.core.metadata".format(LOG_PREFIX))
+
+# Valid characters for names
+# this allows only numbers and letters for first char, 
+# then numbers, letters, and underscore for the rest
+_valid_chars_re = "^[a-zA-Z][a-zA-Z0-9_]*$"
```

Comment:
Should it be renamed? `safe_name_chars_re` maybe?

Reply:
I'll make it a function in core.utils named validate_name. Treat it like validate_item. I do something similar in core.triggers, where I'm removing invalid characters. I'll try to work it in there. I'll get to this in about an hour.

Reply:
Good idea for a function, but in triggers you replace or remove invalid chars. Here I'm raising a value error. Will it have both functions? If it replaces we can't use it here because the namespace will be different than the user expects. It will always be translated to the same thing by the validate function, but if we ever modify the function the rules won't work anymore.

Reply:
I did move the validation function to core.utils, but ended up removing all validation from this module and added try/except. Java is already validating everything and will throw the exceptions, so no need to recreate that here.